### DISCUSSION TITLE
Remove --no-print-directory from recursive make calls

### DISCUSSION THREAD

**Discussion by btgould** [premake/premake-core#1665] [Q&A] [answered]
I am using premake to generate gnu makefiles for my project. By default, premake specifies `--no-print-directory` at the location of each recursive make call. 

e.g. The makefile generated in the root directory has several entries (one for each project in my workspace) each of the form:
```
Project:
ifneq (,$(Project_config))
	@echo "==== Building Project ($(Project_config)) ===="
	@${MAKE} --no-print-directory -C path/to/project/dir -f Makefile config=$(Project_config)
endif
```
And each corresponding subdirectory has a makefile that handles building that specific project. 

I would like to remove the `--no-print-directory` option from the recursive make calls. Specifically, this is so that the `makefile-tools` extension for vscode can properly parse my makefiles and generate intellisense information, which it needs printed information each time make changes directories to do. 

Is there an option that I could specify somewhere to remove the `--no-print-directory` from these calls? 

Environment: Ubuntu 20.04 using premake5 version 5.0.0-alpha15

**Comment by starkos** [comment, marked as answer]
There is not an official way to do that currently, and not a particularly clean way to override the logic either. You could try to override `premake.make.projectrules()` (or `premake.gmake2.projectrules` depending on which action you're using), and just modify the logic to suit. Something like (untested):

```lua
premake.override(premake.make, 'projectiles', function(base, wks)
    for prj in p.workspace.eachproject(wks) do
        local deps = project.getdependencies(prj)
        deps = table.extract(deps, "name")
        _p('%s:%s', p.esc(prj.name), make.list(deps))

        local cfgvar = make.tovar(prj.name)
        _p('ifneq (,$(%s_config))', cfgvar)

        _p(1,'@echo "==== Building %s ($(%s_config)) ===="', prj.name, cfgvar)

        local prjpath = p.filename(prj, make.getmakefilename(prj, true))
        local prjdir = path.getdirectory(path.getrelative(wks.location, prjpath))
        local prjname = path.getname(prjpath)

        -- removed `--no-print-directory` here --
        _x(1,'@${MAKE} -C %s -f %s config=$(%s_config)', prjdir, prjname, cfgvar)

        _p('endif')
        _p('')
    end
end)
```

This won't be future-proof though; it would be better to submit a PR to add an option to control it.

**Reply by btgould** [reply]
Thank you! With some minor modifications, this worked! 

Unfortunately, I am not too familiar with premake's inner workings, so I am not sure what would even need to be changed in a PR to fix this issue. 

If anyone else is coming here with the same question, the only required changes were to add some local declarations: 
```
local p = premake
local gmake2 = p.modules.gmake2
...
premake.override(gmake2, 'projectrules', function(base, wks)
    local project = p.project
    ...
end
```
The project local must be declared inside the override scope so that it does not interfere with premake's default `project` keyword. I also changed all instances of `make` to `gmake2`, simply because that is the action I am using.
